src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -89,23 +104,12 @@ public List<NuGetPackageInfo> NuGetPackages
 
         public IProgressEvents Run()
         {
-            List<string> languages = new List<string>();
-            if (this.projectSystemHelper.GetSolutionManagedProjects().Any(p => ProjectSystemHelper.IsCSharpProject(p)))
-            {
-                languages.Add(SonarQubeServiceWrapper.CSharpLanguage);
-            }
-
-            if (this.projectSystemHelper.GetSolutionManagedProjects().Any(p => ProjectSystemHelper.IsVBProject(p)))
-            {
-                languages.Add(SonarQubeServiceWrapper.VBLanguage);
-            }
-
-            Debug.Assert(languages.Count > 0, "Expecting managed projects in solution");
             Debug.Assert(this.host.ActiveSection != null, "Expect the section to be attached at least until this method returns");
+            Debug.Assert(this.projectSystem.GetSolutionProjects().Any(), "Expecting projects in solution");
 
             IProgressEvents progress = ProgressStepRunner.StartAsync(this.host,
                 this.host.ActiveSection.ProgressHost,
-                (controller) => this.CreateWorkflowSteps(controller, languages));
```

Comment:
The projects/languages to bind with can no longer determined before communicating with the server (due to the Test project regex property). A new discovery step has been added to determine which projects (if any) in the solution are applicable for binding.